merkle/stack.go

```diff
@@ -0,0 +1,172 @@
+package merkle
```

Comment:
I think this code only pertains to the RHP; let's put it in `net/rhp`

Reply:
should change `merkle.NodeHash` to be a passthrough to `blake2b.SumPair`, though.